src/include/dav1d/dav1d.rs

```diff
@@ -175,11 +177,22 @@ impl TryFrom<Dav1dSettings> for Rav1dSettings {
             decode_frame_type,
             reserved: _,
         } = value;
+        validate_input!((n_threads >= 0 && n_threads <= 256, Rav1dError::EINVAL))?;
+        validate_input!((
+            max_frame_delay >= 0 && max_frame_delay <= 256,
+            Rav1dError::EINVAL
+        ))?;
+        validate_input!((
+            operating_point >= 0 && operating_point <= 31,
+            Rav1dError::EINVAL
+        ))?;
         Ok(Self {
-            n_threads,
-            max_frame_delay,
+            n_threads: InRange::<u16, 0, 256>::new(n_threads.try_into().unwrap()).unwrap(),
```

Comment:
```suggestion
            n_threads: InRange::new(n_threads.try_into().unwrap()).unwrap(),
```

Does this work?  If so, can you do this for all of the other `InRange::new`s, too?  So we just have the bounds in the field type.